### DISCUSSION TITLE
Is jax optimized for m1 chips on macos?

### DISCUSSION THREAD

**Discussion by machineko** [jax-ml/jax#5084] [Q&A] [answered]
As tensorflow 2.4 is optimized for m1 is this the case for jax also [blogpost](https://blog.tensorflow.org/2020/11/accelerating-tensorflow-performance-on-mac.html )

**Comment by hawkinsp** [comment, marked as answer]
Not at the moment. But we'd welcome contributions that did so!

(It's not entirely clear to me what APIs Apple are exposing and whether we can usefully target them from XLA.)

**Reply by machineko** [reply]
We should probably ask some Tensorflow folks about this :P

**Comment by 8bitmp3** [comment]
Looping in @rxwei cc @skye @jekbradbury

**Comment by 8bitmp3** [comment]
New PR: `make native bazel build work on Apple Silicon` #47594 https://github.com/tensorflow/tensorflow/pull/47594

**Reply by akbir** [reply]
Update: This is merged!

**Comment by 8bitmp3** [comment]
Update: issue https://github.com/google/jax/issues/5501#issuecomment-878700978 and PR https://github.com/google/jax/pull/7268 from @hawkinsp

**Reply by 8bitmp3** [reply]
https://github.com/google/jax/issues/5501#issuecomment-885964303 by @moellenh:

> Update2: It runs now! :)

**Reply by 8bitmp3** [reply]
https://github.com/google/jax/issues/5501#issuecomment-906737202 by @dfm 👍 . Workflow: https://github.com/dfm/custom-wheels/blob/217352d8dcc311080dacd91f9919d913cdb9f0d2/.github/workflows/jaxlib.yml#L11-L49
